src/components/charts/list/DiscoverCharts.tsx

```diff
@@ -364,43 +391,42 @@ function DiscoverChartList() {
-                                                                        showDescription={!isGrid}
-                                                                        subtractChart={subtractChart}
-                                                                        onClick={(chartId) =>
-                                                                            state.charts.length === 0
-                                                                                ? history.push(
-                                                                                      `${url}/chart/${chart.id}`,
-                                                                                  )
-                                                                                : selectChart(chartId)
-                                                                        }
-                                                                    />
-                                                                ))}
-                                                        </div>
+                                                        <>
+                                                            <ChartListHeader charts={state.charts} />
+                                                            <div className={`chart-grid ${!isGrid ? 'list-view' : ''}`}>
+                                                                {chartList
+                                                                    .slice(0, showDeployModal ? 12 : chartList.length)
+                                                                    .map((chart) => (
+                                                                        <ChartSelect
+                                                                            key={chart.id}
+                                                                            chart={chart}
+                                                                            selectedCount={
+                                                                                state.selectedInstances[chart.id]
+                                                                                    ?.length
+                                                                            }
+                                                                            showCheckBoxOnHoverOnly={
+                                                                                state.charts.length === 0
+                                                                            }
+                                                                            addChart={addChart}
+                                                                            showDescription={!isGrid}
+                                                                            subtractChart={subtractChart}
+                                                                            onClick={(chartId) =>
```

Comment:
extract this function

Reply:
After extracting, for every selection api for icon hits which reload the icon